mne/viz/_figure.py

```diff
@@ -298,12 +294,15 @@ def _load_data(self, start=None, stop=None):
             # subtract one sample from tstart before searchsorted, to make sure
             # we land on the left side of the boundary time (avoid precision
             # errors)
-            ix = np.searchsorted(self.mne.boundary_times,
-                                 self.mne.t_start - self.mne.sampling_period)
-            item = slice(ix, ix + self.mne.n_epochs)
+            ix_start = np.searchsorted(self.mne.boundary_times,
+                                       self.mne.t_start -
+                                       self.mne.sampling_period)
+            ix_stop = ix_start + self.mne.n_epochs
+            item = slice(ix_start, ix_stop)
             data = np.concatenate(self.mne.inst.get_data(item=item), axis=-1)
-            times = np.arange(len(self.mne.inst) * len(self.mne.inst.times)
-                              )[start:stop] / self.mne.info['sfreq']
+            times = np.arange(self.mne.boundary_times[ix_start],
```

Comment:
@drammock In some cases (with testing-data), `times` shape greatly diverted from data. With this solution, sometimes an off-by-one-error has to be corrected by the index but otherwise it seems to work with all tests and manual inspection. What were your experiences with `times` here? Would you approve this solution?

Reply:
> With this solution, sometimes an off-by-one-error has to be corrected

I would expect this because the float arithmetic in `arange` with a `float` step (or bounds) will be inexact.

Did you try instead something like `times = np.arange(ix_start, ix_stop) / self.mne.info['sfreq']`? That uses int arithmetic in `arange` so the number of elements is deterministic (i.e., shouldn't have any off-by-one error).

Based on the old code I also would have expected `np.arange(start, stop)` / self.mne.info['sfreq']` to be good enough. But I'm not familiar with the `boundary_times` stuff to know which is better or more correct here...

Reply:
Thank you, you're right there shouldn't be a difference. Knowing this I checked the error from the tests again and in fact found a little bug in pyqtgraph-backend that caused the problem when changing duration.